### ISSUE TITLE
Support 100 Continue

### ISSUE THREAD

**Issue by aseemk** [httpie/cli#152]
Hey there,

Really great work on HTTPie. I've come to love it a lot -- so much nicer than cURL. =)

Just chiming in w/ a feature request: it'd be great if HTTPie supported `100 Continue`:

http://www.w3.org/Protocols/rfc2616/rfc2616-sec8.html

You probably already know what this is, but if not, a client can send an `Expect: 100-continue` request header, giving the server a chance to respond either with `100 Continue` or a different, final status code -- before the client sends along the request body.

E.g. Amazon S3 recommends using this:

http://docs.aws.amazon.com/AmazonS3/latest/dev/RESTRedirect.html

I've tried passing along this header myself, which "works" -- but HTTPie doesn't actually take that into account at all; it still sends the request body.

https://gist.github.com/aseemk/6116887

Thanks much for the consideration!

**Comment by spencergibb** [comment]
BTW https://github.com/psf/requests/issues/713 is closed

**Comment by sigmavirus24** [comment]
@spencergibb correct. There's nothing that library or urllib3 can do to support this until Python's standard library stops ignoring/swallowing it whole and hiding it from them

**Comment by Almad** [comment]
Still blocked by https://bugs.python.org/issue1346874
